tests/scenarios/v2-addon-dev-test.ts

```diff
@@ -0,0 +1,209 @@
+import path from 'path';
+import { appScenarios } from './scenarios';
+import { PreparedApp, Project } from 'scenario-tester';
+import QUnit from 'qunit';
+import merge from 'lodash/merge';
+import execa from 'execa';
+
+const { module: Qmodule, test } = QUnit;
+
+appScenarios
+  .map('v2 addon can have imports of template-only components', async project => {
+    let packageJSON = {
+      keywords: ['ember-addon'],
+      exports: {
+        './*': './dist/*',
+        './addon-main.js': './config/addon-main.cjs',
+      },
+      files: ['dist', 'config/addon-main.cjs'],
+      'ember-addon': {
+        type: 'addon',
+        version: 2,
+        main: './config/addon-main.cjs',
+        'app-js': {},
+      },
+      devDependencies: {
```

Comment:
This is an unreliable way to add dependencies, you don't need to edit package.json manually, use `project.linkDevDependency` instead.

Reply:
updated